hy/compiler.py

```diff
@@ -2689,18 +2689,20 @@ def hy_compile(tree, module_name, root=ast.Module, get_expr=False):
     body = []
     expr = None
 
-    if tree:
+    if not (isinstance(tree, HyObject) or type(tree) is list):
```

Comment:
Is this really that useful? IMO part of the charm of Python is duck-typing, where someone can pass a custom but API-compatible list subclass, and it "just works". Can't you just do `isinstance(tree, list)`?